Require Python 3

tavinathanson opened the pull request:
No longer support Python 2, including:
* No longer test with Python 2.
* No longer use compatibility layers for Py2/3.
* Remove version checking logic.
* Clarify on README.
* Add `python_requires` constraint in `setup.py`.

Also update 2016 to 2017 in files.

I figured we should go all out and not allow Python 2 usage at all if we're no longer testing with Python 2. Otherwise, issues like `str` vs. `unicode` resulting in the wrong number of patients in the `Cohort` will be highly likely.

jburos commented on cohorts/cohort.py:
```diff
@@ -576,9 +573,6 @@ def _load_single_patient_variants(self, patient, filter_fn, use_cache=True, **kw
             filter_fn_name = self._get_function_name(filter_fn)
             logger.debug("loading variants for patient {} with filter_fn {}".format(patient.id, filter_fn_name))
             use_filtered_cache = use_cache
-            if sys.version_info < (3, 3):
```
Given our requirement is for python >=3 we should probably still check for versions 3.1 and 3.2?

jburos replied:
and/or, add these versions to our test matrix?

tavinathanson replied:
Added 3.3, 3.4 and 3.6 to our matrix, and requiring >= 3.3 to use Cohorts.

jburos commented on setup.py:
```diff
@@ -67,6 +67,7 @@
         ],
         install_requires=install_requires,
         dependency_links=dependency_links,
+        python_requires=">=3",
```
See above comment re: dependency on version >= 3.3

tavinathanson commented:
@jburos I think I addressed your comments, lmk!

tavinathanson commented:
@jburos turns out `pysam` `0.9.0`, which we `conda install` in Travis (https://github.com/hammerlab/cohorts/pull/49#issuecomment-225646603), requires 3.4.

So, instead, I made it `>= 0.9.0` and also added `pysam` to our `requirements.txt` (since we do directly use it now). I also removed `3.3` from Travis, since no current version of `pysam` on `Conda` appears to support `3.3`.

Hopefully that'll all work.
